theories/datatypes/List.ec

```diff
@@ -1998,6 +2044,36 @@ case: (0 <= n < size s) => ?; last by rewrite !put_out 1?size_map.
 by rewrite !put_in 1?size_map // !(map_cons, map_cat, map_take, map_drop).
 qed.
 
+lemma drop_putK (s : 'a list) (i j : int) (x : 'a) :
+     j < i 
+  => drop i (put s j x) = drop i s.
+proof.
+elim: s i j => [ * | x' s ih i j gtj_i]; 1: by rewrite put_empty. 
+case (i <= 0) => [le0_i | /ltzNge gt0_i]; 1: by rewrite put_out /#.
+case (j = 0) => [-> /# | neq0_j].
+by rewrite put_consS // /= ih /#.
+qed.
+
+lemma drop_putC (s : 'a list) (i j : int) (x : 'a) :
+     0 <= i <= j 
+  => drop i (put s j x) = put (drop i s) (j - i) x.
+proof.
+elim: s i j => [ * | x' s ih i j [ge0_i gei_j] /=]; 1: smt(put_empty).
+case (i = 0) => [-> /# | neq0_i].
+by rewrite put_consS 1:/# /= (: ! i <= 0) 1:/# /= ih /#.
+qed.
```

Comment:
How would you state that lemma? With a nested `if then else`? Thing is that for `take_put`, the two cases are each other's negations, while for `drop_put` this would not be the case I think. Specifically, `i` needs to be greater or equal to 0 for the `drop_putC` to hold. Negating the precondition of `drop_putC` would give `i < 0 \/ j < i`, meaning that, if I would want to do combine the `drop` lemmas with a single `if then else` statement, I would need to prove that the consequent of `drop_putK` holds with only `i < 0` as a precondition, which I think is not the case. What I can prove is something like the following:
```
lemma drop_put (s : 'a list) (i j : int) (x : 'a):
  drop i (put s j x) = if 0 <= i <= j 
                       then put (drop i s) (j - i) x 
                       else if i < 0 
                            then put s j x
                            else drop i s.
```